### PR TITLE
Removing IDocumentAttributes.branch everywhere

### LINKED ISSUE #6889: Can we remove IDocumentAttributes.branch?
As far as I remember most of the code relying on it was removed. So it's time to figure out a plan on how to finally remove it.
Note that definition is in protocol-definitions, so there will be some complexity validating that indeed we can remove it.

### PR DESCRIPTION
**Pull request by chensixx**
Fixes #6889 
These are the removal of IDocumentAttributes.branch in client and server. As well as the version-bump. Will be merged after #8213 and a pre-release

### CODE REVIEW

File: common/lib/protocol-definitions/src/storage.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export interface IDocumentAttributes {
     /**
      * Name of the branch that created the snapshot
      */
-    branch: string;
```
Comment (pleath) [inline common/lib/protocol-definitions/src/storage.ts:10]: Could you remove the commented field?

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 2]
```diff
@@ -322,7 +322,6 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // fix the branch name in attributes
         // this prevents issues when generating summaries
         const documentAttributes: api.IDocumentAttributes = JSON.parse(bufferToString(blob, "utf8"));
-        documentAttributes.branch = this.documentId;
         const content = JSON.stringify(documentAttributes);
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:325]: you should be able to remove this whole block, as the only reason it exists is to patch branch info

File: common/lib/container-definitions/package-lock.json [hunk 3]
```diff
@@ -596,7 +596,7 @@
     },
     "@fluidframework/protocol-definitions": {
       "version": "0.1026.0-42695",
-      "resolved": "https://registry.npmjs.org/@fluidframework/protocol-definitions/-/protocol-definitions-0.1026.0-42695.tgz",
+      "resolved": "https://registry.npmjs.org/@fluidframework/protocol-definitions/-/protocol-definitions-0.1026.0-43898.tgz",
```
Comment (anthony-murphy) [inline common/lib/container-definitions/package-lock.json:599]: this is wrong. the sha should change when the version does re-install in this package

File: server/routerlicious/packages/services-shared/src/storage.ts [hunk 4]
```diff
@@ -60,7 +60,7 @@ export class DocumentStorage implements IDocumentStorage {
         values: [string, ICommittedProposal][],
     ): ISummaryTree {
         const documentAttributes: IDocumentAttributes = {
-            branch: documentId,
+            // branch: documentId,
```
Comment (jatgarg) [inline server/routerlicious/packages/services-shared/src/storage.ts:63]: Why commenting out here and not removing like at other places?

Comment (chensixx) [reply server/routerlicious/packages/services-shared/src/storage.ts:63]: sorry, was just testing the pipelines, gonna remove now

**Review by jatgarg — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 5]
```diff
@@ -318,14 +316,7 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         if (!this.attributesBlobHandles.has(blobId)) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:316]: You could have removed this bloc as well (as it's noop), and likely - tracking of attributesBlobHandles, as it's no longer needed.

Comment (chensixx) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:316]: oh i see!

**Pull request merged**